### PR TITLE
Add Rat role

### PR DESCRIPTION
**Pull request by hthanhdev**

### CODE REVIEW

File: json/text_template.json [hunk 1]
```diff
@@ -1388,5 +1388,41 @@
...
+        "💡 Example: `{bot_prefix}bite 2`",
+        "⚠️ You can use this skill anytime at night. Target yourself if you change your mind."
+      ]
+    }
+  },
+  "rat_kill_text": {
+    "params": ["target"],
+    "template": {
+      "vi": ["🐭🐺 Chuột ma sói muốn cắn {target} trong đêm nay"],
+      "en": ["🐭🐺 Rat {werewolf} wants to bite {target} tonight"]
```
Comment (nitro2) [inline json/text_template.json:1411]:
```suggestion
      "en": ["🐭🐺 Rat werewolf wants to bite {target} tonight"]
```

File: test.py [hunk 2]
```diff
@@ -100,16 +100,16 @@ async def test_game():
...
-            await test_case(game, path)
-        else:
-            continue
+    # directory = "testcases"
+    # for filename in os.listdir(directory):
+    #     if filename.endswith(".json"):
+    #         path = os.path.join(directory, filename)
+    #         await test_case(game, path)
+    #     else:
+    #         continue
```
Comment (nitro2) [inline test.py:112]: Remember to revert this in final version

File: testcases/case-rat-bite-fox.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+				"r1 bite f1"
+			]
+		},
+		{
+			"time": "day2",
+			"alive": [
+				"v1",
+				"r1",
+				"wi1"
+			],
```
Comment (nitro2) [inline testcases/case-rat-bite-fox.json:42]: Rat will die after biting fox

File: game/roles/rat.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    def get_bite_target(self):
+        return self.bite_target
+    
+    def set_bite_target(self, target_id):
+        self.bite_target = target_id
+
+    def register_bite_target(self, target_id):
+        self.set_bite_target(target_id)
+
+        return text_templates.generate_text("rat_kill_text", target=f"<@{target_id}>")
```
Comment (nitro2) [inline game/roles/rat.py:23]: I think you don't need to create new methods. We already have `register_target` in character.py. You can reuse that without creating new methods.

File: game/roles/rat.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+from game.roles.villager import Villager
+from game.roles.werewolf import Werewolf
+
+
+class Rat(Villager):
```
Comment (nitro2) [inline game/roles/rat.py:5]: Should inherit from Character?

**Review by nitro2 — DISMISSED** [review]

File: game/roles/rat.py (hunk 5 above)
Comment (neihousaigaai) [reply game/roles/rat.py]: As we have discussed before, I modified Rat inherits from Character.

**Review by nitro2 — APPROVED** [review]

**Pull request merged**